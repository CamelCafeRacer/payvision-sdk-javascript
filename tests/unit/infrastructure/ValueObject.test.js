const ValueObject = require('../../../src/infrastructure/ValueObject')

describe('ValueObject', () => {
  let vo
  beforeEach(() => {
    vo = new ValueObject()
  })

  describe('validates', () => {
    it('required', () => {
      expect(() => vo.importValue('testVal', 123, { type: 'type', required: true })).not.toThrow()
      expect(() => vo.importValue('testVal', 123, { type: 'type', required: false })).not.toThrow()
      expect(() => vo.importValue('testVal', undefined, { type: 'type', required: false })).not.toThrow()
      expect(() => vo.importValue('testVal', undefined, { type: 'type', required: true })).toThrowErrorMatchingSnapshot()
    })

    it('type string', () => {
      expect(() => vo.importValue('testVal', '123', { type: 'string', required: true })).not.toThrow()
      expect(() => vo.importValue('testVal', '', { type: 'string', required: true })).not.toThrow()
      expect(() => vo.importValue('testVal', 123, { type: 'string', required: true })).toThrowErrorMatchingSnapshot()
      expect(() => vo.importValue('testVal', true, { type: 'string', required: true })).toThrowErrorMatchingSnapshot()
      expect(() => vo.importValue('testVal', true, { type: 'string', required: true })).toThrowErrorMatchingSnapshot()
    })
  })
})
