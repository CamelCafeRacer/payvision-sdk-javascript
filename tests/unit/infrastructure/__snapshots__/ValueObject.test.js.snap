// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ValueObject validates required 1`] = `"Property [testVal] is required on [ValueObject]"`;

exports[`ValueObject validates type string 1`] = `"Property [testVal] expected a string, got [number] [123] on [ValueObject]"`;

exports[`ValueObject validates type string 2`] = `"Property [testVal] expected a string, got [boolean] [true] on [ValueObject]"`;

exports[`ValueObject validates type string 3`] = `"Property [testVal] expected a string, got [boolean] [true] on [ValueObject]"`;
